### ISSUE TITLE
Graph merger

### ISSUE THREAD

**Issue by skreutzer** [csensemakers/lib-protocol-experiments#5]
Build a graph merger, which

- creates new nodes if they don't already exist
- creates new edges if they don't already exist
- edges are created not targeting the nodes of the source, but the already existing nodes in the destination

**Comment by ronentk** [comment]
What are example inputs for the graph merger? The Library JSON format?

**Comment by skreutzer** [comment]
Any graph-oriented arrangement. Could/should support multiple of those, ideally the different ones are just a different "front-end", or maybe there's a common graph interface the merger supports/implements.

Practically speaking, the "splitter" turning Library JSON previously into some graph pseudo-format I made up (splitter's "back-end"), I rewrote that for the aggregator to not produce the graph pseudo-format, but to interface with the merger.

This way, the merger in/for the aggregator is currently generic/agnostic about the source format, because doesn't know nor is aware of the source semantics. Currently this is obviously too primitive, because eventually some source semantics should be represented in the graph database and aggregator in order to support graph operations, but would assume we rather try to research ways to represent source semantics in some generic way inside the graph, vs. making a domain-specific/rigid one-off single-case graph/database (then for every different source type, also likely having a hard time relating/mixing/correlating different domains/sources). Depends on cost/benefit which is better, simply something to investigate/explore.
